### ISSUE TITLE
homography() results depend on order of points

### ISSUE THREAD

**Issue by shie0041** [arrayfire/arrayfire#1264]
Below is my post from the Google groups page with details on this issue.  It seems that the results of the homography() function depend on the order of the X and Y features passed in.  This should not be the case.

-Eric

I'm trying to use harris() and homography() to do image registration.  I start with an array called imgBase which is a 150x150 array of floats on the range [0,255].  I can plot this in ArrayFire and it looks correct (assuming I normalize it to the range [0,1]).  The image is a portion of a satellite image (part of Matlab's westconcordaerial.png) and has some nice corner features from buildings, roads, etc.

To try to understand what is going on, I've done the following:

af::features featuresBase = af::harris(imgBase);
af::features featuresReg = af::harris(imgBase);
af::homography(H_ctr, inliers, featuresReg.getX(), featuresReg.getY(), featuresBase.getX(), featuresBase.getY(), AF_HOMOGRAPHY_RANSAC, 5.0f, 1000, f32);

The featuresBase and featuresReg objects each contain 434 features.  These features are identical except that they are ordered differently.  I do not know why they would be ordered differently as they are passed the same image.  I've checked this property of the features by exporting them to Matlab.  When plotted on a 2D plot they overlay exactly.  However, element-wise differences in X or Y coordinates are not identically zero and I can visually inspect that the vectors are not equal.

The homography() call then reports only 10 inliers.  It returns the homography

```
0.1973    0.0222    0.0010
```

   -0.0693   -0.0947    0.0009
  -20.4484   -5.5688   -0.3293

which makes little sense.  Does the ordering of the X, Y points affect the RANSAC result?  It seems like it should not.

If I change the homography() function call to:

af::homography(H_ctr, inliers, featuresBase.getX(), featuresBase.getY(), featuresBase.getX(), featuresBase.getY(), AF_HOMOGRAPHY_RANSAC, 5.0f, 1000, f32);

the function behaves as expected.  It returns 434 inliers and a homography array of

```
0.1970   -0.0000   -0.0000
0.0000    0.1970    0.0000
```

   -0.0000   -0.0000    0.1970

I expected this to be the 3x3 identity matrix, but I assume I should normalize by the lower-right-most element.

**Comment by pavanky** [comment]
@shie0041 can

**Comment by pavanky** [comment]
@shie0041 can you check if the latest devel fixes this issue?

**Comment by shie0041** [comment]
Pavan,

Thanks for the heads up.  I’ll have to take a look at this next week as I’m in some training courses all week through Friday.

Best regards,
Eric Shields

**Comment by pavanky** [comment]
@shie0041 we just released v3.4.0 that should contain the fix. You need not build it from source.
